src/main/java/org/folio/rest/helper/FundsHelper.java

```diff
@@ -100,23 +99,37 @@ public FundsHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
       });
   }
 
-  private CompletableFuture<String> getCurrentFiscalYearId(String ledgerId) {
+  public CompletableFuture<FiscalYear> getCurrentFiscalYear(String ledgerId) {
     FiscalYearsHelper fiscalYearsHelper = new FiscalYearsHelper(httpClient, okapiHeaders, ctx, lang);
+    return getTwoFirstFiscalYears(ledgerId, fiscalYearsHelper)
+      .thenApply(twoFirstFiscalYears -> {
+        if(CollectionUtils.isNotEmpty(twoFirstFiscalYears)) {
+          if(twoFirstFiscalYears.size() > 1 && isOverlapped(twoFirstFiscalYears.get(0), twoFirstFiscalYears.get(1))) {
+            return twoFirstFiscalYears.get(1);
```

Comment:
say the 2 FYs are 
1. FY19 : Jan 1 2019- Dec 25th 2019
2. FY20: Jan 1 2020 - Dec 25th 2020
and current date = Dec 26th 2019

According to 1st point in the story:
> (The year with the closest start date that is greater than the current date and is the next year in the series sequence)
the year that is greater than the current date must be returned which is FY 20.  But in this case FY 19 will be returned and isOverlapping will return false

Reply:
In this case `twoFirstFiscalYears` will contain only  FY 20, because  FY 19 is filtered by `SEARCH_CURRENT_FISCAL_YEAR_QUERY`

Reply:
Hi @KVupp,
I believe your example cannot happens because I use the following query for fiscal years selecting:
`"series==\"%s\" AND periodEnd>%s sortBy periodStart"`.
In this case FY19 will not be returned in the FY collection at all because it will be filtered by condition periodEnd>now(). That is, we cut off all the years that have already ended by now. This is correct only for 1st point. Here we can find the following cases:
FY20 : Jan 1 2020 - Dec 25th 2020
FY21: Jan 1 2021 - Dec 25th 2021
and current date = Dec 26th 2019.
In this case FY20 will be returned as its start period is the closest to current date.